### PR TITLE
global: update miscellaneous docs for May release [+]

### LINKED ISSUE #140: config: document rate limiting and 4.0 related changes in documentation
We should document 

1) how to configure the rate limiting to not hit it when importing large amounts of records.

Either part of 1 or separate issue for:
2) How to perform bulk imports of larger number of records.

### LINKED ISSUE #168: fix "role" field documentation
Package version (if known): April release

The `role` field is documented as follows in the API:

image

but `role` is actually not part of `person_or_org` it is outside i,e. on same level as `person_or_org` in the implemenation.

Assuming the documentation is wrong here, but worth double-checking @lnielsen ? 

In any case, the `role` field will soon be updated to be a vocabulary field as well.

### PR DESCRIPTION
**Pull request by fenekku**
- resource_types: update new structure
- rights: update to reflect present structure
- migrate: document rate limit via small migrate page
    * That page could be developed over time.
- rest: mention `-k` option
- fixtures: document custom data loading
- metadata: fix role field reference
- closes #168
- closes #140

### CODE REVIEW

**Review by ppanero — APPROVED** [review]
LGTM, few minors and doubts.... I am honestly not sure about changing the structure and mixing the "extensions" into "develop", but let's try it out and see how it goes (cuz I also agree on having less top level menus)

Nice work!! 🚀

File: docs/develop/develop.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+and then re-start the instance.
+
+The command takes care of cleaning the assets for you. This is convenient for
+two reasons, first the module might have already been installed from upstream and
+therefore some links already exist, or these files already exist (as hard copies) if the
+installation was not done using the `--development` flag. As you can see, many
+things can go wrong, so the added time is worth it.
+
```
[the remark this answers is not in the document]
Comment (fenekku) [reply docs/develop/develop.md:35]: You mean the Jinja templates, right? I've added a quick sentence or 2 about it :+1:

File: docs/install/migrate.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "records.create": "25000 per hour;500 per minute",
+    "records.publish": "25000 per hour;500 per minute",
+    "draft_files.create": "25000 per hour;500 per minute",
+    "draft_files.update_content": "25000 per hour;500 per minute",
+    "draft_files.create_commit": "25000 per hour;500 per minute"
+}
+```
+
+Naturally, choose a rate that makes sense for you!
+
```
[the remark this answers is not in the document]
Comment (fenekku) [reply docs/install/migrate.md:33]: removed!

File: docs/install/migrate.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+Then you will want to POST to the API to create the records, but InvenioRDM rate limits the API usage. By default, we don't want our API to be abused or subject to denial-of-service attacks. For the period of time when records are being ingested, however, you will want to allow a greater request rate.
+
+## Rate limiting
+
+The Flask-Limiter library is used to control this via its [configuration options](https://flask-limiter.readthedocs.io/en/stable/#configuration). There you will find a range of configuration to adapt your strategy to your needs.
+
+For our purposes, a couple of simple options are available:
+
+### Option 1: RATELIMIT_PER_ENDPOINT
```
[the remark this answers is not in the document]
Comment (fenekku) [reply docs/install/migrate.md:16]: Ah yes, I was originally looking for something like this but there was no string notation for it. So I outlined those approaches. And on a fresh take this morning I discovered: `RATELIMIT_ENABLED` . Thanks!

File: docs/reference/rest_api.md [hunk 4]
```diff
@@ -15,15 +15,19 @@ The only authentication method supported at the moment for REST API calls is by
 **Authorization HTTP header (recommended)**
 
 ```shell
-curl -H "Authorization: Bearer API-TOKEN" https://127.0.0.1:5000/api/records
+curl -k -H "Authorization: Bearer API-TOKEN" https://127.0.0.1:5000/api/records
```
[the remark this answers is not in the document]
Comment (fenekku) [reply docs/reference/rest_api.md:18]: Thanks to @ines-cruz on this one for noticing :smile:

**Comment by fenekku** [comment]
The merging of "Develop" and "Extensions" was suggested by @lnielsen during last standup. The idea of creating a "Use" section was also mentioned, but I didn't do it yet because more changes are needed. Overall, there is more thinking to be done about how to structure the docs for sure.

**Pull request merged**